tests/store/artifact/utils/test_model_utils.py

```diff
@@ -33,6 +33,12 @@ def test_parse_models_uri_with_version(uri, expected_name, expected_version):
         ("models:/AdsModel1/pROduction", "AdsModel1", "pROduction"),  # case insensitive
         ("models:/Ads Model 1/None", "Ads Model 1", "None"),
         ("models://scope:key@databricks/Ads Model 1/None", "Ads Model 1", "None"),
+        (
+            "models:/Name/Stage@Alias",
+            "Name",
+            "Stage@Alias",
+        ),  # technically allowed, but the backend would throw
```

Comment:
Why are these allowed if they don't start with `models:`? Shouldn't this throw here: https://github.com/mlflow/mlflow/pull/8322/files#diff-29023164bda459e390ab6a88de6785d3a9712fc0dcef29a4f20ce7c7d7d2cb62R57-R58